File: ui/react/src/action/RemoveAsset.ts
import SheetAction from "./SheetAction";
import Action from "./Action";

class RemoveAsset extends SheetAction {
    public assetName: string;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const assetName: string = input.assetName;
        return new RemoveAsset(sheetName, assetName);
    }

    constructor(sheetName: string, assetName: string) {
        super(sheetName);
        this.assetName = assetName;
    }

    public wrapper(): Action {
        const action = new Action();
        action.removeAsset = this;
        return action;
    }

}

export default RemoveAsset;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/RemoveAsset.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;

public class RemoveAsset extends SheetAction implements Action {
    private String assetName;

    public RemoveAsset() {
    }

    public RemoveAsset(String sheetName, String assetName) {
        super(sheetName);
        this.assetName = assetName;
    }

    public String getAssetName() {
        return assetName;
    }

    public void setAssetName(String assetName) {
        this.assetName = assetName;
    }
}
